strapi/strapi
# replace axiosInstance with useFetchClient inside the i18n plugin

### What does it do?

## Replace every instance of axiosInstance with the new useFetchClient and getFetchClient in the i18n plugin

We have already implemented the new constructs useFetchClient and getFetchClient in[ this PR](https://github.com/strapi/strapi/pull/14759)

Now we need to replace every place where we use axiosInstance with the new constructs, test the possible regression errors and still provide the old implementation of the axiosInstance into the helper plugin with a message of the removal in v5

### Why is it needed?

Inside the Strapi project we have a lot of different implementations of the same code to use axios and its axiosInstance. In this branch we replace axiosInstance in the users-permissions plugin

### How to test it?

You need to use the new construct inside every axiosInstance call and check the possible regression errors

List of the API calls in the i18n where we use the new construct and how to test them:
Here you can see a list of API calls changed with the instructions to test them:

- CMEditViewCopyLocale handleConfirmCopyLocale GET ex: http://localhost:1337/content-manager/collection-types/api::category.category/1 from the Admin click  on “Settings“, under the “GENERAL“ section, then click on “Internationalization“ under “GLOBAL SETTING“, add a new Locale, then inside the Content Manager select a collection and add a new entry, Save, and then select another locale from the Dropdown and click "Fill in from another locale", confirm your choice and the API call will be triggered, as you can see in this video https://www.loom.com/share/30317c40d48b4413a12612213e611333
